src/kaleidoscope/device/Base.h

```diff
@@ -0,0 +1,504 @@
+/* -*- mode: c++ -*-
+ * kaleidoscope::device::Base -- Kaleidoscope device Base class
+ * Copyright (C) 2017, 2018, 2019  Keyboard.io, Inc
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file kaleidoscope/device/Base.h
+ * Base class for Kaleidoscope device libraries.
+ */
+
+#pragma once
+
+#include "kaleidoscope/MatrixAddr.h"
+#include "kaleidoscope_internal/deprecations.h"
+#include "kaleidoscope/macro_helpers.h"
+
+#include "kaleidoscope/driver/keyscanner/None.h"
+#include "kaleidoscope/driver/led/None.h"
+#include "kaleidoscope/driver/mcu/None.h"
+#include "kaleidoscope/driver/bootloader/None.h"
+#include "kaleidoscope/driver/storage/None.h"
+
+#ifndef CRGB
+#error cRGB and CRGB *must* be defined before including this header!
+#endif
+
+/* All hardware libraries must define the following types and macros:
+ * kaleidoscope::Device - a typedef to your device's class.
+ * CRGB(r,g,b) - explained below
+ * cRGB, a structure with at least three members: r, g, and b -
+ * compilation will fail otherwise.
+ *
+ * Despite its name, the members do not need to be in the order r g b -- most
+ * likely they will be in an order that is convenient for the hardware. So
+ * initializing a cRGB with a struct literal will give surprising results for any
+ * colors where r, g, and b do not have the same value. Each Hardware library
+ * defines a CRGB(r,g,b) macro which returns a literal cRGB with the given values.
+ */
+
+namespace kaleidoscope {
+namespace device {
+
+struct BaseProps {
+  typedef kaleidoscope::driver::keyscanner::BaseProps KeyScannerProps;
+  typedef kaleidoscope::driver::keyscanner::None KeyScanner;
+  typedef kaleidoscope::driver::led::BaseProps LEDDriverProps;
+  typedef kaleidoscope::driver::led::None LEDDriver;
+  typedef kaleidoscope::driver::mcu::None MCU;
+  typedef kaleidoscope::driver::bootloader::None Bootloader;
+  typedef kaleidoscope::driver::storage::BaseProps StorageProps;
+  typedef kaleidoscope::driver::storage::None Storage;
+};
+
+template<typename _DeviceProps>
+class Base {
+ private:
+  class DummySerial {
+   public:
+    DummySerial() {}
+  };
+
+  static DummySerial dummy_serial_;
```

Comment:
We're using 'None' to describe other noop classes. We use 'Noop' for NoOp key addresses. 
We don't use 'Dummy' anywhere else in code, just in comments.

NoSerial? NoopSerial?

Reply:
`NoopSerial` it is then, I like that better than `NoSerial`.